samples/csharp/chatapp/SampleChatHub.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Azure.Messaging.WebPubSub;
+using Microsoft.Azure.WebPubSub.AspNetCore;
+using Microsoft.Azure.WebPubSub.Common;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace chatapp
+{
+    public class SampleChatHub : WebPubSubHub
+    {
+        private readonly WebPubSubServiceClient _serviceClient;
+
+        public SampleChatHub(WebPubSubServiceClient serviceClient)
+        {
+            _serviceClient = serviceClient;
+        }
+
+        public override ValueTask<WebPubSubEventResponse> OnConnectAsync(ConnectEventRequest request, CancellationToken cancellationToken)
+        {
+            // not register event will never be triggered.
+            throw new NotImplementedException();
```

Comment:
Is this abstract or virtual?

Reply:
Abstract, cause we don't know default returns in connect/message events. If customer doesn't register these events in the `Event Handler` settings and this method will never be triggered in server side, otherwise, they should implement this.

Reply:
This isn't good. We should make this virtual if it's not always required.

Reply:
I see. Basically we suggest customer to implement the registered events. And agree this is not required and we should make it same behavior as not register. Will make virtual and return null as response body. Same for the `OnMessageRecievedAsync()`.